src/ui/public/kbn_top_nav/bread_crumbs/bread_crumb_urls.js

```diff
@@ -14,7 +14,7 @@
  */
 export function getBreadCrumbUrls(breadcrumbs, url) {
   return breadcrumbs.map(breadcrumb => {
-    const breadCrumbStartIndex = url.toLowerCase().lastIndexOf(breadcrumb.toLowerCase());
+    const breadCrumbStartIndex = url.toLowerCase().lastIndexOf(breadcrumb.replace(/\W/, '').toLowerCase());
```

Comment:
Thanks for the explanation. I think we should do two things here:

1. Maybe break apart this long chain of calls into multiple lines with multiple `const`s that have meaningful names. I think that will help with readability and explain what's going on here.

2. I think you need to change `/\W/` to `/\W/g` in order to replace all non-word characters in `breadcrumb`, not just the first one. That way if we ever have a breadcrumb with more than two words in it, this code will continue to work correctly. For example:

```js
> "quick brown fox".replace(/\W/, '')
'quickbrown fox'
> "quick brown fox".replace(/\W/g, '')
'quickbrownfox'
```